generic_expression_patterns_modules/process.py

```diff
@@ -203,3 +232,92 @@ def generate_summary_table(
     summary.to_csv(summary_file, float_format="%.5f", sep="\t")
 
     return summary
+
+
+def merge_ranks_to_compare(
+    your_summary_gene_ranks_df,
+    reference_gene_ranks_file,
+    reference_gene_name_col,
+    reference_rank_col,
+):
+    """
+    Given dataframes of your ranking of genes and reference ranking
+    of genes. This function merges the ranking into one dataframe
+    to be able to compare, `shared_gene_rank_df`
+
+    Arguments
+    ---------
+    your_summary_gene_ranks_df: df
+        dataframe containing your rank per gene
+    reference_gene_ranks_file: file
+        file contining reference ranks per gene
+    reference_gene_name_col: str
+        column header containing the reference genes
+    reference_rank_col: str
+        column header containing the reference rank
+
+    Returns
+    -------
+    Dataframe containing your ranking and the reference ranking per gene
+
+    """
+    # Read in reference gene ranks file
+    reference_gene_ranks_df = pd.read_csv(reference_gene_ranks_file, header=0, sep="\t")
+
+    # Get list of our genes
+    gene_ids = list(your_summary_gene_ranks_df.index)
+
+    # Get list of published generic genes
+    published_generic_genes = list(reference_gene_ranks_df[reference_gene_name_col])
+
+    # Get intersection of gene lists
+    shared_genes = set(gene_ids).intersection(published_generic_genes)
+
+    # Get rank of shared genes
+    your_gene_rank_df = pd.DataFrame(
+        your_summary_gene_ranks_df.loc[shared_genes, "Rank (simulated)"]
+    )
+
+    # Merge published ranking
+    shared_gene_rank_df = pd.merge(
+        your_gene_rank_df,
+        reference_gene_ranks_df[[reference_rank_col, reference_gene_name_col]],
+        left_index=True,
+        right_on=reference_gene_name_col,
+    )
+
+    return shared_gene_rank_df
+
+
+def scale_reference_ranking(merged_gene_ranks_df, reference_rank_col):
+    """
+    In the case where the reference ranking and your ranking are not
+    in the same range, this function scales the reference ranking
+    to be in the range as your ranking.
+
+    For example, if reference ranking ranged from (0,1) and your
+    ranking ranged from (0,100). This function would scale the
+    reference ranking to also be between 0 and 100.
+
+    Note: This function is assuming that the reference ranking range
+    is smaller than yours
+
+    Arguments
+    ---------
+    merged_gene_ranks: df
+        dataframe containing your rank and reference rank per gene
+    reference_rank_col: str
+        column header containing the reference rank
+
+    Returns
+    -------
+    The same merged_gene_ranks dataframe with reference ranks re-scaled
+    """
+    # Scale published ranking to our range
+    max_rank = max(merged_gene_ranks_df["Rank (simulated)"])
```

Comment:
Can you use [scikit-learn's MinMaxScaler](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.MinMaxScaler.html) for this? It seems like that would be more general.

You'd need to test whether that gives the same results as what you're doing, though - I'm not sure off the top of my head.

Reply:
Looks like this is doing more of a z-scoring, which isn't what we want in this case

Reply:
I think it should be a linear scaling to a range with a particular min and max value (the scikit-learn version of z-scoring is StandardScaler). It implements [this formula](https://math.stackexchange.com/a/43699), which I think is the general version of what you're doing.

Anyway, just something to think about for the future. If what you're doing is working fine, no reason to change it now.

Reply:
AH...you're correct! I misread what was happening. I will definitely change this in the next PR because it will make the pipeline more robust